pkg/storage/kubernetes/ipam.go

```diff
@@ -35,6 +35,9 @@ func NewKubernetesIPAM(containerID string, ipamConf whereaboutstypes.IPAMConfig)
 		return nil, err
 	} else if ctx, ok := cfg.Contexts[cfg.CurrentContext]; ok && ctx != nil {
 		namespace = ctx.Namespace
+		if namespace == "" {
+			namespace = metav1.NamespaceSystem
+		}
```

Comment:
actually , would you mind exporting this into a function ? 

Something like

```golang
func wbNamespace(ctx *clientcmdapi.Context) string {
	namespace := ctx.Namespace
	if namespace == "" {
		return metav1.NamespaceSystem
	}
	return namespace
}
```

Reply:
I think that's wrong. There are 2 deployment paths for whereabouts: 
- a simple CNI plugin - which relies on having a kubeconfig path - i.e. it doesn't have an in cluster config
- a controller / cronjob running in a pod, which rely on the in cluster config.

The above are exclusive: the developer should be explicit and know which to call. By doing this change, you are imo mixing these two types of clients, thus making it more confusing, and hard to reason about.